opm/parser/eclipse/EclipseState/Schedule/Compsegs.cpp

```
@@ -0,0 +1,136 @@
+            } else {
+                // TODO: the end of the previous connection or range
+                // 'previous' should be in term of the input order
+                // since basically no specific order for the completions
+                throw std::runtime_error("this way to obtain DISTANCE_START not implemented yet!");
+            }
+            if (record->getItem("DISTANCE_END")->hasValue(0)) {
+                distance_end = record->getItem("DISTANCE_END")->getRawDouble(0);
+            } else {
+                // TODO: the distance_start plus the thickness of the grid block
+                throw std::runtime_error("this way to obtain DISTANCE_END not implemented yet!");
+            }
+
+            WellCompletion::DirectionEnum direction;
+            if (record->getItem("DIRECTION")->hasValue(0)) {
+                direction = WellCompletion::DirectionEnumFromString(record->getItem("DIRECTION")->getString(0));
+            } else if (!record->getItem("DISTANCE_END")->hasValue(0)) {
+                throw std::runtime_error("the direction has to be specified when DISTANCE_END in the record is not specified");
+            }
+
+            int end_IJK;
+            if (record->getItem("END_IJK")->hasValue(0)) {
+                // following the corrdinate rule for completions
+                end_IJK = record->getItem("END_IJK")->getInt(0) - 1;
+                if (!record->getItem("DIRECTION")->hasValue(0)) {
+                    throw std::runtime_error("the direction has to be specified when END_IJK in the record is specified");
+                }
+            } else {
+                // only one completion is specified here
+                end_IJK = -1;
+            }
+
+            double center_depth;
+            if (record->getItem("CENTER_DEPTH")->hasValue(0)) {
+                center_depth = record->getItem("CENTER_DEPTH")->getRawDouble(0);
+            } else {
+                center_depth = 0.;
```

Hmmm - `center_depth == 0` does not seem to be a valid value? Unless simulation will continue happily with `center_depth == 0` I really think you should raise an exception here.